api/api.go

```diff
@@ -130,10 +131,108 @@ func (a *API) ExecuteTask(serviceID, taskKey string, inputData map[string]interf
 	if err = a.execDB.Save(exec); err != nil {
 		return "", err
 	}
-	go a.ps.Pub(exec, s.TaskSubscriptionChannel())
+	go a.ps.Pub(exec, s.ExecutionSubTopic())
 	return exec.ID, nil
 }
 
+// ListenEvent listens events matches with eventFilter on serviceID.
+func (a *API) ListenEvent(service string, f *EventFilter) (*EventListener, error) {
+	s, err := a.db.Get(service)
+	if err != nil {
+		return nil, err
+	}
+
+	if f != nil && f.Key != "" {
+		if _, err := s.GetEvent(f.Key); err != nil {
+			return nil, err
+		}
+	}
+
+	l := NewEventListener(a.ps, s.EventSubTopic(), f)
+	go l.Listen()
+	return l, nil
+}
+
+// ListenExecution listens executions on service.
+func (a *API) ListenExecution(service string, f *ExecutionFilter) (*ExecutionListener, error) {
+	s, err := a.db.Get(service)
+	if err != nil {
+		return nil, err
+	}
+
+	if f != nil {
+		if f.TaskKey == "" && f.OutputKey != "" {
```

Comment:
any reason to force this? For now we are able to listen to any `OutputKey`

Reply:
is it make sense to listen for any `OutputKey` without `TaskKey`? What such listening would mean?

I can change it but it seemed to me rather like a bug not a feature ;)

Reply:
I used it in some cases when I know that the service always returns the output `error` then I can only listen to that `outputKey` and catch all errors

Reply:
This behavior is normal (I used it in an old version of the core). We can have outputFilter only if taskFilter is present, that makes more sense but we just need to update the documentation https://github.com/mesg-foundation/core/issues/922

Can you revert this modification and sorry for that

Reply:
done